### PR TITLE
Recommend more idiomatic Devise practices

### PR DESCRIPTION
**Pull request by wagenet**
Based on information provided here https://gist.github.com/josevalim/fb706b1e933ef01e4fb6.
This also allows for other Devise recommendations to be followed.

### CODE REVIEW

File: packages/ember-simple-auth-devise/README.md [hunk 1]
```diff
@@ -57,17 +57,16 @@ needed the format handling can be left out of course_):
...
-      format.html { super }
-      format.json do
-        self.resource = warden.authenticate!(auth_options)
-        sign_in(resource_name, resource)
+    super do |user|
+      if request.format.json?
         data = {
-          token:      self.resource.authentication_token,
-          user_email: self.resource.email
+          user_token: user.authentication_token,
```
Comment (marcoow) [inline packages/ember-simple-auth-devise/README.md]: This was renamed to just `token` in Ember Simple Auth's last release as [`ActionController::HttpAuthentication::Token`](http://api.rubyonrails.org/classes/ActionController/HttpAuthentication/Token.html) expects the token to be sent as `Authorization: Token token=…` and we wanted to have a consistent name for the token property that's used throughout the entire library.

**Comment by marcoow** [comment]
:+1:

We have a linter configured that [breaks on trailing spaces though](https://travis-ci.org/simplabs/ember-simple-auth/jobs/50689022#L120)...

**Comment by wagenet** [comment]
Corrected both issues.

**Comment by marcoow** [comment]
:+1:

**Pull request merged**